External/SPMmodified/xASL/xASL_io_Nifti2Im.m

```diff
@@ -64,7 +70,12 @@
             niiMat = true; % load mat below
         else
             % Load NIfTI and JSON sidecar
-            [nii, ~, json] = xASL_io_ReadNifti(niftiIn);
+			if nargout < 2
```

Comment:
It might look complicated, but it's the correct thing to do and makes no sense to do it otherwise. 
If the user doesn't request json, then it makes no sense to waste resources in reading it.

This is not about catching errors or initialization, it is about skipping the calculation of certain things that the user didn't request. This makes the code more faster and less time complex, which is especially helpful for image reading that takes unnecessary resources. Or 90% of cases of NII reading doesn't require the JSON. It's just an optimization of code without any adverse effects of caveats